codec-http2/src/main/java/io/netty/handler/codec/http2/AbstractHttp2StreamChannel.java

```
@@ -0,0 +1,304 @@
+     * the channel (using Unsafe) instead of notifying handlers of the message with {@code
+     * channelRead()}. Additional inbound messages must not arrive after this one.
+     */
+    protected static final Object CLOSE_MESSAGE = new Object();
+    private static final ChannelMetadata METADATA = new ChannelMetadata(false, 16);
+    private static final int MAX_READER_STACK_DEPTH = 8;
+
+    private final ChannelConfig config = new DefaultChannelConfig(this);
+    private final Queue<Object> inboundBuffer = new ArrayDeque<Object>();
+    private final Runnable fireChildReadCompleteTask = new Runnable() {
+        @Override
+        public void run() {
+            if (readInProgress) {
+                readInProgress = false;
+                unsafe().recvBufAllocHandle().readComplete();
+                pipeline().fireChannelReadComplete();
+            }
+        }
+    };
+
+    private boolean closed;
+    private boolean readInProgress;
+
+    public AbstractHttp2StreamChannel(Channel parent) {
+        super(parent);
+    }
+
+    @Override
+    public ChannelMetadata metadata() {
+        return METADATA;
+    }
+
+    @Override
+    public ChannelConfig config() {
+        return config;
+    }
+
+    @Override
+    public boolean isOpen() {
+        return !closed;
+    }
+
+    @Override
+    public boolean isActive() {
+        return !closed;
+    }
+
+    @Override
+    protected AbstractUnsafe newUnsafe() {
+        return new Unsafe();
+    }
+
+    @Override
+    protected boolean isCompatible(EventLoop loop) {
+        return true;
+    }
+
+    @Override
+    protected SocketAddress localAddress0() {
+        return parent().localAddress();
+    }
+
+    @Override
+    protected SocketAddress remoteAddress0() {
+        return parent().remoteAddress();
+    }
+
+    @Override
+    protected void doBind(SocketAddress localAddress) throws Exception { }
```

as bind is not supported we may want to throw an exception ?